spring-data-jdbc/src/test/java/org/springframework/data/jdbc/testing/AuditableNamedParameterJdbcTemplate.java

```diff
@@ -0,0 +1,296 @@
+package org.springframework.data.jdbc.testing;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import java.util.stream.Stream;
+
+import org.assertj.core.api.Assertions;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.dao.DataAccessException;
+import org.springframework.jdbc.core.JdbcOperations;
+import org.springframework.jdbc.core.PreparedStatementCallback;
+import org.springframework.jdbc.core.ResultSetExtractor;
+import org.springframework.jdbc.core.RowCallbackHandler;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcOperations;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.jdbc.support.rowset.SqlRowSet;
+
+/**
+ * {@link NamedParameterJdbcOperations} implementation for tests, that is capable to audit the
+ * executed queries. The actual execution is delegated to {@link #delegate}.
+ *
+ * @author Mikhail Polivakha
+ */
+public class AuditableNamedParameterJdbcTemplate implements NamedParameterJdbcOperations {
```

Comment:
I can see the motivation where this originates. Let's not sneak in new test utils that aren't fully leak-proof in a bugfix PR.

Reply:
So, am I correct that the suggestion is to make `AuditableNamedParameterJdbcTemplate` class as `static` nested class inside `JdbcRepositoryIdGenerationIntegrationTests`? 

Or the suggestion is to re-think the design of the class in general? The motivation is to track the actual statement sent to the storage.

Reply:
I would remove it entirely. Using `DataSource` decoration should give us a direct way to capture and assert statements instead of trying to decorate a higher-level component